pkg/tracker/tracker.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright © 2023 SUSE LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package tracker implements a tracking mechanism to keep track
+// of the ports during various container event types e.g start, stop
+package tracker
+
+import (
+	"context"
+	"net"
+
+	"github.com/docker/go-connections/nat"
+)
+
+// NetTracker is the interface that wraps the methods
+// that are used to manage Add/Remove tcp listeners.
+type NetTracker interface {
+	// AddListener creates a TCP listener for a given IP and Port
+	// it returns an error if it fails.
```

Comment:
```suggestion
	// AddListener creates a TCP listener for a given IP and Port;
	// it returns an error if it fails.
```
Alternatively,
```suggestion
	// AddListener creates a TCP listener for a given IP and Port.
	// It returns an error if it fails.
```
Or even better,
```suggestion
	// AddListener creates a TCP listener for a given IP and Port.
```
(Also applies to `RemoveListener`.)